rfds/0000-expanding-software-ids.md

```diff
@@ -0,0 +1,484 @@
+CVE records to publish potentially time-sensitive information about a new
+vulnerability, but are potentially unable to populate CPE data until after the
+initial CVE record publication.
+
+Mechanical applicability determinations—especially searches of CVE data based
+on software identifiers—are compromised if the searcher cannot rely on the
+identifiers to be available when and where they are needed.
+
+Moreover, some vulnerability conditions cannot be expressed adequately using
+CPE. For example, sometimes a vulnerability is only present when certain
+modules or files are present, but CPEs do not capture software at the module or
+file level. To put it another way, CPE is a relatively coarse-grained software
+identifier, identifying software “products,” potentially constrained with
+version information, but not components or materials within those software
+products. While the `affected` array provides fields to constrain applicability
+at a greater level of granularity, the `cpeApplicability` object does not, and
+there is substantial ambiguity in interpreting the `cpes` field within the
+`affected` array's product objects, which was part of the motivation for the
+introduction of the `cpeApplicability` structure.
+
+CPEs are also not used universally across different software ecosystems. Open
+source software projects are generally less well represented in the
+NIST-maintained CPE dictionary than closed source software. This means sole
+reliance on CPE as the mechanism for identifying software within the CVE record
+format leaves CVE less able to identify open source software affected by a
+vulnerability.
+
+As for the `affected` array's "identifier-like" fields, there are two variants
+to assess. First, a `vendor` and `product` pair; second, a `collectionURL` and
+`packageName`.
+
+For the `vendor` and `product` pair, they are potentially useful for human
+interpreters, but are of limited value for automated applicability matching.
+They express similar information to what is found within a CPE, but because
+they are not constrained to use well-known terms defined within a central
+dictionary, there is substantial risk of divergence, where the terms used by
+a CNA to identify the vendor or product may not match the terms used by
+downstream consumers to search for or match against records to determine
+applicability.
+
+For the `collectionURL` and `packageName` pair, these function similarly to a
+Package URL, but again lack a well-defined construction to ensure consistency
+and easy automated cross-referencing between datasets. The `collectionURL`
+field provides many example, but is not constrained beyond requiring a valid
+URI. The `packageNamem` field is entirely unconstrained. Similarly to the first
+option for identifying affected products, they are most useful for human
+matching of CVE records, but not for automated processing of applicability at
+scale.
+
+## Proposed Solution
```

Comment:
In terms of "alternatives considered"...

Why not address the lack of convention to the `vendor`/`product` and/or `collectionURL`/`packageName` fields rather than proposing adding a whole new field?

e.g. `vendor` and `product` could by convention be required to map the vendor and product of a CPE, and then be  able to be validated against the CPE Dictionary. The expectations for `collectionURL` and `packageName` could similarly be defined.